apps/iris/plots.py

```diff
@@ -0,0 +1,253 @@
+"""Module for generating plots of the Iris data set.
+"""
+import sys
+
+import data
+import pandas as pd
+import plotly.express as px
+import plotly.graph_objects as go
+import plotly.subplots as psp
+
+
+def _generate_scatterplot(iris_df: pd.DataFrame) -> go.Figure:
+    figure = psp.make_subplots(
+        rows=1,
+        cols=2,
+        subplot_titles=["Sepal Width vs. Length", "Petal Width vs. Length"],
+    )
+    sepal_plot = px.scatter(
+        data_frame=iris_df,
+        x="sepal_width",
+        y="sepal_length",
+        color="class",
+    )
+    petal_plot = px.scatter(
+        data_frame=iris_df, x="petal_width", y="petal_length", color="class"
+    )
+
+    # Stack exchange question 60633891 really helped here.
+    for trace in sepal_plot.data:
+        figure.add_trace(trace, 1, 1)
+    for trace in petal_plot.data:
+        figure.add_trace(trace, 1, 2)
+
+    # Stack exchange 58849925 helped here understanding editing Figures.
+    figure.layout["xaxis"]["title"] = "sepal width (cm)"
+    figure.layout["yaxis"]["title"] = "sepal length (cm)"
+    figure.layout["xaxis2"]["title"] = "petal width (cm)"
+    figure.layout["yaxis2"]["title"] = "petal length (cm)"
+    figure.layout["title"] = "Scatterplots of Iris Features"
```

Comment:
Legend has duplicate entries for each subplot. I saw some people using go.Scatter with the showlegend=False option. Maybe that will work, if you have not already tried it.

Reply:
Thanks for the suggestion, I was able to remove the duplicate entries with this in `051a889`.